packages/snaps-execution-environments/src/common/BaseSnapExecutor.ts

```diff
@@ -258,9 +260,12 @@ export class BaseSnapExecutor {
     if (!isValidJson(requestObject) || !isObject(requestObject)) {
```

Comment:
In some of the places where you've added sanitisation, you've left `isValidJson`'s in. At this point the sanitise is doing that validation, so you could remove them.

Reply:
Removed `isValidJson` where `sanitizeJsonStructure` is used.